llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -560,6 +573,27 @@ transform_ext::StructuredOpMatcher::input(int64_t position,
   return *this;
 }
 
+transform_ext::StructuredOpMatcher &
+transform_ext::StructuredOpMatcher::input(int64_t position,
+                                          CaptureElementType elem) {
+  predicates.push_back([=](linalg::LinalgOp linalgOp) -> bool {
+    LLVM_DEBUG(DBGS() << "input operand #" << position
+                      << " capture element type");
+    int64_t updatedPosition = position;
+    if (!makeValidPositiveIndex(updatedPosition, linalgOp.getNumDpsInputs()))
+      return false;
+    auto shapedType = linalgOp.getDpsInputOperand(updatedPosition)
+                          ->get()
+                          .getType()
+                          .dyn_cast<ShapedType>();
+    if (!shapedType)
+      return false;
```

Comment:
Nit: it may be helpful to have a debug print here saying "not a shaped type" so we know when the matcher failed.